services/openvpn/service/manager.go

```diff
@@ -264,50 +266,77 @@ func (m *Manager) ProvideConfig(_ string, sessionConfig json.RawMessage) (*sessi
 	}
 	vpnConfig.Ports = []int{} // TODO This line will not be required once we will have unique VPN config for every connection.
 
-	// Older clients do not send any sessionConfig, but we should keep back compatibility and not fail in this case.
-	if sessionConfig != nil && len(sessionConfig) > 0 && m.natPinger.Valid() {
-		var consumerConfig openvpn_service.ConsumerConfig
-		err := json.Unmarshal(sessionConfig, &consumerConfig)
-		if err != nil {
-			return nil, errors.Wrap(err, "could not parse consumer config")
-		}
-
-		if len(consumerConfig.Ports) == 0 {
-			cp, err := m.natPingerPorts.Acquire()
+	// TODO this backward compatibility block needs to be removed once we will fully migrate to the p2p communication.
+	if conn == nil {
+		// Older clients do not send any sessionConfig, but we should keep back compatibility and not fail in this case.
```

Comment:
I think all clients that are now able to connect to our new nodes sends this config. We can remove this check.

Reply:
Removed it.